src/ripple/test/jtx/impl/Account.cpp

```diff
@@ -36,15 +44,28 @@ Account::Account(std::string name,
 {
 }
 
-Account::Account (std::string name, KeyType type)
-    : name_(std::move(name))
+Account::Account(std::string name, KeyType type)
 {
-    auto const keys = generateKeyPair(
-        type, generateSeed(name_));
-    pk_ = keys.first;
-    sk_ = keys.second;
-    id_ = calcAccountID(pk_);
-    human_ = toBase58(id_);
+    name_ = std::move(name);
+    auto const p = std::make_pair(name_, type);
+    auto const iter = cache_.find(p);
+    if (iter == cache_.end())
+    {
+        auto const keys = generateKeyPair(
+            type, generateSeed(name_));
+        pk_ = keys.first;
+        sk_ = keys.second;
+        id_ = calcAccountID(pk_);
+        human_ = toBase58(id_);
+        cache_[p] = *this;
+    }
+    else
+    {
+        pk_ = iter->second.pk_;
+        sk_ = iter->second.sk_;
+        id_ = iter->second.id_;
+        human_ = iter->second.human_;
+    }
```

Comment:
We can simplify this code by using a delegating to the move constructor and a helper function that returns the account from the cache & add it to the cache if needed. Code: https://github.com/ripple/rippled/commit/4ea8c5617bef844e78ea6fdd8de61b6f8d461891

Note: That code required making one of the private constructors public, so I could efficiently emplace into a map. If we need to keep that constructor private I'll modify this solution.